### PR TITLE
A06: As Admin, I would like to add / update / delete useful resources and links.

### PR DESCRIPTION
**Pull request by irinatu17**
## Description
Admin CRUD operations feature in the resources page: 
- Only authenticated admin can add/edit/delete resources. **Defensive design** is in place redirecting other users to the home page with the corresponding messages, when attempted to enter URL manually.
- Add/edit/delete buttons are visible **only to admin**.
- All admin actions(add/edit/delete) are followed by the corresponding messages.
- Clicking **Add Resource** button opens the Add Resource page, where admin can fill out the Name, URL and Description fields and the new resource will be visible in the Resources table.
- Clicking a **Delete** button opens a modal, asking an admin to confirm the deletion, resource will be permanently removed from the database.
- Clicking **Edit** button redirects a user to the Edit Resource page with prepopulated fields. 

## Pull request type

- [ ] #M01 (Miscellaneous User Story #01)
- [ ] #P02 (Participant User Story #02)
- [ ] #S03 (Staff User Story #03)
- [X] #A06 (Admin User Story) - A06: As Admin, I would like to add / update / delete useful resources and links.
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue

#25 
#34 

## Testing
- Manual tests were implemented to test the feature functionality, everything is working as expected. Admin CRUD operations were tested from the superuser's account. Attempts to use admin's functionality from other accounts and by non-authenticated users were made to test defensive design.
- Responsiveness testing was implemented using Google Chrome's developer tools, everything looks as expected across all the different screen sizes.
- Browser compatibility was checked on Chrome, Edge and FireFox browsers.

## Screenshots
image

## Does this introduce a breaking change

- [ ] Yes
- [X] No

### CODE REVIEW

File: resources/forms.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class ResourceForm(forms.ModelForm):
+    """Allows admin to add/edit resources"""
+
+    class Meta:
+        model = Resource
+        fields = '__all__'
+
+
```
Comment (stefdworschak) [inline resources/forms.py:12]: As per PEP8 method definitions inside a class are surrounded by a single blank line. Can you please remove the extra blank line?

File: resources/views.py [hunk 2]
```diff
@@ -9,3 +13,68 @@ def resources(request):
...
+        if form.is_valid():
+            resource = form.save()
+            messages.success(request, 'Successfully added resource!')
+            return redirect(reverse('resources'))
+        else:
+            messages.error(request, 'Failed to add resource. Please ensure the form is valid.')
+    else:
+        form = ResourceForm()
+
+    template = 'resources/add_resource.html'
```
Comment (stefdworschak) [inline resources/views.py:36]: Not sure you need to declare a variable for this, I would just pass it directly to the `render` function.

File: resources/views.py [hunk 3]
```diff
@@ -9,3 +13,68 @@ def resources(request):
...
+            form.save()
+            messages.success(request, 'Successfully updated resource!')
+            return redirect(reverse('resources'))
+        else:
+            messages.error(request, 'Failed to update resource.\
+                 Please ensure the form is valid.')
+    else:
+        form = ResourceForm(instance=resource)
+
+    template = 'resources/edit_resource.html'
```
Comment (stefdworschak) [inline resources/views.py:74]: Same here.

**Review by stefdworschak — APPROVED** [review]
LGTM

Just look at my few comments and let me know when I can merge this request.

**Comment by irinatu17** [comment]
@stefdworschak I've done those few changes, you can have a look now

File: resources/forms.py (hunk 1 above)
Comment (irinatu17) [reply resources/forms.py]: Sure, think I've done this

File: resources/views.py (hunk 2 above)
Comment (irinatu17) [reply resources/views.py:36]: I've moved `ResourceForm()` to the `render` function,  is that what you mean here?

**Pull request merged**